### DISCUSSION TITLE
Merchant NPCs spawned using Events fail to load the shop for the location

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#14605] [Ideas and suggestions]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Spawning a merchant NPC using an event will result in an empty shop if the location doesnt already have a merchant

![image](https://github.com/user-attachments/assets/7a82fad8-bfe3-4e0c-9bb9-c882d25aeb5a)

**The outpost generation parameters need to have a merchant defined to begin with** for the shops to load, otherwise merchants spawn with an empty shop menu.

![image](https://github.com/user-attachments/assets/cd8dccca-7970-4362-b91d-2b67df471132)

_The problem arose while trying to circumvent the lack of configurable starter outpost..._

_Also it would be a good thing to allow merchants to use a default "store" value (defined in locationtypes.xml) in case of for example merchants spawned in level transitions (say you want to spawn a merchant on a beacon or a shuttle)_

### Reproduction steps

Use this event to test:

1. Remove merchants from Outpost configuration so that the outpost spawns without an existing merchant
2. Start new campaign
3. `triggerevent startermerchant`
4. Empty inventory
5. Add merchant to outpost configuration again
6. Restart campaign
7. `triggerevent startermerchant`
8. Merchant has inventory again

```xml
    <!-- deletes the existing merchant and spawns a new one (this works) -->
    <ScriptedEvent identifier="startermerchant" commonness="100">
      <!-- <TagAction criteria="humanprefabidentifier:merchantoutpost" tag="removenpc" ContinueIfNoTargetsFound="true" />
      <StatusEffectAction targettag="removenpc">
        <StatusEffect type="Always" target="Character" oneshot="true">
          <RemoveCharacter />
        </StatusEffect>
      </StatusEffectAction> -->
      <SpawnAction npcsetidentifier="outpostnpcs1" npcidentifier="merchantoutpost" targettag="startermerchant" spawnlocation="Outpost" spawnpointtag="shop_outpost" teamid="FriendlyNPC" />
    </ScriptedEvent>
```

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.5.9.2 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
I would consider this a feature request: stores are currently generated per-location, and cannot be created "on the fly" like this. I agree though that it'd be very useful to be able to create them using events, I think we might look into this in the future.

**Reply by ubertpendragon** [reply]
The weird thing is, if outpost has a shop NPC defined in outpostgenerationparameters.xml to begin with, you can remove the merchant NPC and spawn a new one, the shop  will work.

Only if there was no NPC defined in the first place you won't see the shop appear. 



Either way this is how much jank I have to go through to circumvent the lack of enforced starter outpost 😔

**Comment by Regalis11** [comment]
This has now (finally) been addressed to support some of the features in the upcoming DLC, fix coming up in the next test build. :)

PR in our private repo: https://github.com/FakeFishGames/Barotrauma-development/pull/8202
